xbmc/guilib/GraphicContext.cpp

```diff
@@ -369,19 +369,19 @@ bool CGraphicContext::IsValidResolution(RESOLUTION res)
 }
 
 // call SetVideoResolutionInternal and ensure its done from mainthread
-void CGraphicContext::SetVideoResolution(RESOLUTION res, bool forceUpdate)
+void CGraphicContext::SetVideoResolution(RESOLUTION res, bool forceUpdate /*= false*/, bool switch3d /*= false*/)
```

Comment:
No. I did originally implement separate 3D resolutions but it didn't work well. You get a huge list of extra resolutions in system/video output. Each 3D format has extra calibration data, so calibrating subtitle height for 1080p24 needs to be done again for 1080p24 SBS and 1080p24 TAB and 1080p25 FP.
In reality the TAB/SBS flags are just out-of-band AV Infoframe signalling. It is just requested as a modifier (https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/egl/EGLNativeTypeRaspberryPI.cpp#L254) to the hdmi mode.

The reason for this flag is we get an unwanted resolution change from the fullscreen message which occurs after playback starts, and so after we believe we are in 3D mode, but before the renderer configure which knows the correct framerate. That means HDMI does:
1080p60 2D -> 1080p60 3D -> 1080p24 3D
and I want a way to ignore the 1080p60 3D.